pandas/io/tests/test_parsers.py

```diff
@@ -3349,6 +3378,12 @@ def test_whitespace_lines(self):
         df = self.read_csv(StringIO(data))
         tm.assert_almost_equal(df.values, expected)
 
+    def test_read_with_bad_header(self):
+        errmsg = "but only \d+ lines in file"
+        with tm.assertRaisesRegexp(StopIteration, errmsg):
```

Comment:
Okay, now I think I understand what you mean by "syncing" them.  Will give that a shot.